package/MDAnalysis/coordinates/PDB.py

```diff
@@ -398,6 +398,8 @@ def _read_frame(self, frame):
                     warnings.warn("Failed to read CRYST1 record, "
                                   "possibly invalid PDB file, got:\n{}"
                                   "".format(line))
+                if (self.ts._unitcell[:3] == np.array([1,1,1],dtype=float)).all():
```

Comment:
ls this somewhat closer to what you're asking?
               `try:
                    cell_check = [line[6:15], line[15:24],
                                  line[24:33], line[33:40],
                                  line[40:47], line[47:54]]
                    if (np.array(cell_check,dtype=float) == 
                        np.array([1,1,1,90,90,90],dtype=float)).all():
                        warnings.warn("1 A^3 CRYST1 record," 
                                      "possibly an EM structure file, setting box-size to 0",stacklevel=2)
                        self.ts._unitcell=np.array([0,0,0,90,90,90],dtype=float)    
                except ValueError:
                    warnings.warn("Failed to read CRYST1 record, "
                                  "possibly invalid PDB file, got:\n{}"
                                  "".format(line))
                else:
                    try:
                        if not (self.ts._unitcell==np.array([0,0,0,90,90,90])).all():
                            self.ts._unitcell[:] = [line[6:15], line[15:24],
                                                    line[24:33], line[33:40],
                                                    line[40:47], line[47:54]]
                    except ValueError:
                        warnings.warn("Failed to read CRYST1 record, "
                                      "possibly invalid PDB file, got:\n{}"
                                      "".format(line))`

Reply:
> Is this what we'd like?

                try:
                    cell_check = [line[6:15], line[15:24],
                                  line[24:33], line[33:40],
                                  line[40:47], line[47:54]]
                    if (np.array(cell_check,dtype=float) == 
                        np.array([1,1,1,90,90,90],dtype=float)).all():
                        warnings.warn("1 A^3 CRYST1 record," 
                                      "possibly an EM structure file, setting box-size to 0",stacklevel=2)
                        self.ts._unitcell=np.array([0,0,0,90,90,90],dtype=float)    
                except ValueError:
                    warnings.warn("Failed to read CRYST1 record, "
                                  "possibly invalid PDB file, got:\n{}"
                                  "".format(line))
                else:
                    try:
                        if not (self.ts._unitcell==np.array([0,0,0,90,90,90])).all():
                            self.ts._unitcell[:] = [line[6:15], line[15:24],
                                                    line[24:33], line[33:40],
                                                    line[40:47], line[47:54]]
                    except ValueError:
                        warnings.warn("Failed to read CRYST1 record, "
                                      "possibly invalid PDB file, got:\n{}"
                                      "".format(line))

Reply:
Just a general point to start with. It is usually easier for us to review if you can push larger code changes to you branch rather than place it in the comments. That way we can get a better feeling for how the code changes fit within the context of the existing code. It also makes it easier for you to read our comments afterwards 😄 

Whilst you have most of the right core elements, I don't think the structure works.

#### `try` block:

What you want to do here is try to extract the lines into a temporary variable (here `cell_dims` might be a more understandable variable name than `cell_check`). There's no point testing/warning for cryo-em default values here because you are already doing this in the `else` block (you don't want to use up extra CPU cycles if you can avoid it).

#### `except` block:

This seems fine.

#### `else` block:

The `else` block is where I'm having the hardest time following what you are doing.
Since you will have already extracted the cell dimensions in the `try` block, there is no point trying to extract them again. Additionally, you have already tested if a `ValueError` will be raised when reading the appropriate parts of `line`. So you don't need the addtiional `try`/`except` here.

In this part of the code I would expect you to be testing the contents of the temporary variable hosting the cell dimensions against the default cryo-em CRYST1 values (i.e. [1,1,1,90,90,90]). If it matches (i.e. is a likely cryo-em structure), then raise a warning. If not then feed the cell dimensions to `self.ts._unitcell`.

Reply:
Thank you so much for taking this. I will keep your points in mind when I revert later.

What I wanted to do is basically have the `cell_dims `variable store the box values which are checked for `CRYST1` values in the first try block. If they turn out to be the same as the specific set of values [1,1,1,90,90,90] then I set the `self.ts._unitcell` as having dimensions of [0,0,0,90,90,90].  And in else block, if the `self.ts._unitcell `is not set as having box-size of 0 then only I go ahead and set it as the cell_dims (again, I should have directly set it as =`cell_dims`). I see that the last try/except was extraneous. I'm really sorry for that.